### PR TITLE
Lower hospital grounds fill priority

### LINKED ISSUE #2162: A hospital area fill overrides a park and a parking fills
A park inside of hospital grounds:
https://www.openstreetmap.org/way/612288154#map=16/47.3777/8.5508
Screenshot_1645377443
Expected: green colour fill same as a patch below selection.

Also a parking there:
Screenshot_1645377480
Expected: pale yellow parking fill

### PR DESCRIPTION
**Pull request by pastk**
Fixes #2162

Made it priority-wise similar to university grounds
https://www.openstreetmap.org/way/612288154#map=16/47.3777/8.5508
hospital

Is some cases it looks less good
https://www.openstreetmap.org/relation/2994307#map=17/56.63908/47.95446
hospital-forest
Also note a strange thing - in 1 & 2 pics a similar patch of forest inside hospital is selected though in one case PP displays forest and in another case - a hospital. And the last screenshot displays a "hospital" still.
Changes to `feature_data.cpp` types list don't help here. Maybe its somehow related to the grounds borders use of multipolygon relation?

A quite nice rendering with symbols in OSM:
hospital-forest

### CODE REVIEW

File: data/styles/clear/include/Icons.mapcss [hunk 1]
```diff
@@ -426,12 +426,10 @@ node[historic=castle],
 area[historic=castle]
 {z-index: 2890;}
 
-node[amenity=hospital],
 node[building=hospital],
 area[building=hospital],
 node[building=university],
 area[building=university],
-area[amenity=hospital],
```
Comment (pastk) [inline data/styles/clear/include/Icons.mapcss:434]: z-index for area fill is set in Basemap.mapcss;
here it was added to raise the priority of icon/text I guess, thus overriding and breaking fill priorities;
also it's impossible to set different z-indices for area and node

File: data/styles/clear/include/Icons.mapcss [hunk 2]
```diff
@@ -1514,6 +1512,10 @@ area|z18-[shop=massage]
 
 /* 5.1 Hospital */
 
+area|z15-[amenity=hospital],
+node|z15-[amenity=hospital],
+{-x-me-text-priority:16000;}
```
Comment (pastk) [inline data/styles/clear/include/Icons.mapcss:1517]: To keep captions priority high (an approach used for university grounds too).

**Comment by biodranik** [comment]
@vng can we sort area objects and draw them from the largest one to the smallest one, on top of each other?

**Comment by pastk** [comment]
This case could be much improved by drawing a hospital's polygon border.
Screenshot_1649433581

**Comment by pastk** [comment]
I would argue that a case of a hospital / university with parkings, lawns, sport pitches etc. is much more common than a hospital / university being surrounded by the woods (and in this case it can have all those smaller area features inside too).

**Comment by vng** [comment]
Seems like we don't understand each other. In term of draw priority P(function):
My point:
P(forest) < P(park) < P(hospital), P(university) < P(garden) < P(parking), P(leisure=pitch)

Your point is that:
P(hospital), P(university) < P(forest), P(park)

Am I right?

**Comment by pastk** [comment]
Parks, parkings and other smaller features should be prioritized higher than hospitals/universities I think.

I agree that generally forests should have a priority lower than hospitals/universities.
Actually forest's priority is strangely high ATM:
https://github.com/organicmaps/organicmaps/blob/14cedf7752cf28c0e5574de3e1e5512d25f38d5a/data/styles/clear/include/Basemap.mapcss#L42-L70
I tried to keep scope of changes in this PR small, its a bit scary to make big changes to priorities it as it may break rendering elsewhere.
A related issue is that `forest` is often used for small strips of trees in urban settings, e.g. https://www.openstreetmap.org/way/273469368 or https://www.openstreetmap.org/way/95514530 (btw note that osm renders them differently compared to a screenshot above).
I think I'll create a separate issue for a more comprehensive areas priority review.

**Comment by vng** [comment]
I found an old test here. There are some questions to it, but I will try to fix my thoughts in this test and we can enhance/discuss according to it.

https://github.com/organicmaps/organicmaps/blob/master/map/style_tests/classificator_tests.cpp#L223

**Comment by pastk** [comment]
> I think I'll create a separate issue for a more comprehensive areas priority review.

Created #2170

**Comment by pastk** [comment]
To recap:
this change makes hospital grounds priority same as university grounds

**Comment by biodranik** [comment]
@vng ?

**Review by vng — APPROVED** [review]

**Pull request merged**